apps/juxtaposition-ui/src/webfiles/web/moderate_user.ejs

```diff
@@ -128,45 +128,231 @@
 					</select>
 				</div>
 				<div class="col">
-					<label for="ban_date">Banned Until:</label>
+					<label for="ban_lift_date" class="labels">Banned Until:</label>
 					<input type="datetime-local" id="ban_lift_date" name="ban_lift_date" value="<%= userSettings.ban_lift_date %>">
 				</div>
 				<div class="col">
-					<label class="labels">Ban Reason</label>
+					<label for="ban_reason" class="labels">Ban Reason</label>
 					<input id="ban_reason" type="text" class="form-control" placeholder="Ban reason" style="width: 100%;" value="<%= userSettings.ban_reason %>">
 				</div>
 			</div>
 			<div class="mt-5 text-center">
 				<button class="btn btn-primary profile-button" type="button" onclick="savePNID()">Save User</button>
 			</div>
 		</div>
-    </div>
+		<div class="mt-5 text-center">
+			<h4 class="text-right">Recently Reported Posts (limit 5)</h4>
+		</div>
+		<ul class="list-content-with-icon-and-text arrow-list" id="news-list-content">
+			<% reports.forEach(function(report) { %>
+				<% let post = postsMap.find((post) => post.id === report.post_id) %>
+				<% if (!post) return; %>
+				<li class="reports">
+					<details>
+						<summary>
+							<div class="hover">
+								<a href="/users/<%= report.reported_by %>" data-pjax="#body" class="icon-container notify">
+									<img src="<%= cdnURL %>/mii/<%= report.reported_by %>/normal_face.png" class="icon">
+								</a>
+								<span class="body messages report" href="#">
+									<span class="text">
+										<a href="/users/<%= report.reported_by %>" class="nick-name">Reported By: <%= userMap.get(report.reported_by) %></a>
+										<span title="<%= moment(report.created_at) %>" class="timestamp"><%= moment(report.created_at).fromNow() %></span>
+									</span>
+									<span class="text">
+										<h4>
+											<% switch (report.reason) {
+											case 0: %>
+											Spoiler
+											<% break;
+											case 1: %>
+											Personal Information
+											<% break;
+											case 2: %>
+											Violent Content
+											<% break;
+											case 3: %>
+											Inappropriate/Harmful Conduct
+											<% break;
+											case 4: %>
+											Hateful/Bullying
+											<% break;
+											case 5: %>
+											Advertising
+											<% break;
+											case 6: %>
+											Sexually Explicit
+											<% break;
+											case 7: %>
+											Piracy
+											<% break;
+											case 8: %>
+											Inappropriate Behavior in Game
+											<% break;
+											case 9: %>
+											Other
+											<% break;
+											case 10: %>
+											Missing Images; Reach out to Jemma with post link to fix
+											<% break;
+											default: %>
+											Unknown
+											<% break; } %>
```

Comment:
I would possibly do this mapping in the code that passes in the `reports` variable - imo there should be minimal logic inside the template for clarity and any data mapping is done before the data is given to the template 

Thoughts?

Reply:
Yeah fair enough. I had copied this from the existing code but it's pretty rough.
There are other places where mapping is done explicitly, just through objects instead of spelt out like this - `communitiesMap`, `userMap`, `postMap` etc. Changing this design is probably a whole different refactor?

Reply:
Maybe we have a `rulesMap` which you give the reason ID to and it has the rule name as the value in the map? That would be a relatively small change, keeping in scope with this PR